test/fiber/test_process.rb

```diff
@@ -0,0 +1,16 @@
+# frozen_string_literal: true
+require 'test/unit'
+require_relative 'scheduler'
+
+class TestFiberProcess < Test::Unit::TestCase
+  def test_system
+    Thread.new do
+      scheduler = Scheduler.new
+      Fiber.set_scheduler scheduler
+
+      Fiber.schedule do
+        system("sleep 1")
```

Comment:
Could you test the value of `$?` after this call?

Also I'd suggest to use `system("true")` (or `echo foo` with backticks) or something which does not take an extra second just for that test.

Reply:
Good idea, I'll update it.

Reply:
Done.

Reply:
Actually it would be good to ensure we have the right Process::Status and not some old one:
```ruby
io = IO.popen("true")
pid = io.pid
io.close
assert_equal pid, $?.pid
```
or simpler:
```ruby
system("exit 17")
assert_equal 17, $?.exitstatus
```